mesonbuild/compilers/compilers.py

```diff
@@ -619,6 +632,7 @@ def __iadd__(self, args: T.Iterable[str]) -> 'CompilerArgs':
         Add two CompilerArgs while taking into account overriding of arguments
         and while preserving the order of arguments as much as possible
         '''
+        this_round_added = {} # a dict that contains a value, when the value was added this round
```

Comment:
This should also be a set. Other than that looks good.

Reply:
Sorry that i did not catch that initially. Fixed now :)